network/f5/bigip_ssl_certificate.py

```diff
@@ -0,0 +1,516 @@
+        )
+
+    def read(self):
+        p = dict()
+        name = self.params['name']
+        partition = self.params['partition']
+
+        if self.key_exists():
+            key = self.api.tm.sys.file.ssl_keys.ssl_key.load(
+                name=name,
+                partition=partition
+            )
+            if hasattr(key, 'checksum'):
+                p['key_checksum'] = str(key.checksum)
+
+        if self.cert_exists():
+            cert = self.api.tm.sys.file.ssl_certs.ssl_cert.load(
+                name=name,
+                partition=partition
+            )
+            if hasattr(cert, 'checksum'):
+                p['cert_checksum'] = str(cert.checksum)
+
+        p['name'] = name
+        return p
+
+    def flush(self):
+        result = dict()
+        state = self.params['state']
+
+        try:
+            if state == "present":
+                changed = self.present()
+            elif state == "absent":
+                changed = self.absent()
+        except iControlUnexpectedHTTPError as e:
+            raise F5ModuleError(str(e))
+
+        result.update(**self.cparams)
+        result.update(dict(changed=changed))
+        return result
+
+    def absent(self):
+        changed = False
+
+        if self.exists():
+            changed = self.delete()
+
+        return changed
+
+    def delete(self):
+        changed = False
+
+        check_mode = self.params['check_mode']
+
+        delete_cert = self.cert_exists()
+        delete_key = self.key_exists()
+
+        if not delete_cert and not delete_key:
+            return changed
+
+        if check_mode:
+            params = dict()
+            params['cert_name'] = name
+            params['key_name'] = name
+            params['partition'] = partition
+            self.cparams = params
+            return True
+
+        tx = self.api.tm.transactions.transaction
+        with TransactionContextManager(tx) as api:
+            if delete_cert:
+                # Delete the certificate
+                c = api.tm.sys.file.ssl_certs.ssl_cert.load(
+                    name=self.params['name'],
+                    partition=self.params['partition']
+                )
+                c.delete()
+                changed = True
+
+            if delete_key:
+                # Delete the certificate key
+                k = self.api.tm.sys.file.ssl_keys.ssl_key.load(
+                    name=self.params['name'],
+                    partition=self.params['partition']
+                )
+                k.delete()
+                changed = True
+        return changed
+
+
+def main():
+    argument_spec = f5_argument_spec()
+
+    meta_args = dict(
+        name=dict(type='str', required=True),
+        cert_content=dict(type='str', default=None),
+        cert_src=dict(type='str', default=None),
+        key_content=dict(type='str', default=None),
+        key_src=dict(type='str', default=None),
+        passphrase=dict(type='str', default=None)
```

Comment:
Can you please add a `no_log` option here, and retest with `-vvv`.

Once that's done I'll merge this